lib/rexml/parsers/baseparser.rb

```diff
@@ -274,8 +274,8 @@ def pull_event
             return [ :elementdecl, @source.match( ELEMENTDECL_PATTERN, true )[1] ]
 
           when ENTITY_START
-            match = @source.match( ENTITYDECL, true ).to_a.compact
-            match[0] = :entitydecl
+            match = @source.match( ENTITYDECL, true )
+            match = match.nil? ? [:entitydecl] : [:entitydecl, *match.captures.compact]
```

Comment:
Do we need `match.nil?` check here?
(Is there any case that the above `@source.match()` failed?)

Reply:
If the match.nil? check is removed, all tests succeed.

But, if the string `<!ENTITY>` comes in, `@source.match()` responds with `nil` and `undefined method ``captures' for nil` is raised.

However, since `<!ENTITY>` violates the XML specification and should be treated as an error, can I remove check `match.nil?`?

https://xml.coverpages.org/xmlBNF.html
```
EntityDecl ::= '<!ENTITY' S Name S EntityDef S? '>'   /* General entities */
	| '<!ENTITY' S '%' S Name S EntityDef S? '>'   /* Parameter entities */
```

Reply:
OK. Could you add a test for the case?

Reply:
Or we can do it as a separated PR.

Reply:
OK
I added [test](https://github.com/ruby/rexml/compare/995d3e2cc133658e1ec2d57b5f1b5ff2fe0ced72..ba9f7fc08de13ab602cfd698295879853fdba0a8)  for this case.